### PR TITLE
test: add scenario for Kubernetes YAML edit feature

### LINKED ISSUE #9259: Implement E2E scenario for the Edit Kube Yaml
blocked by the issue: https://github.com/containers/podman-desktop/issues/9260

### PR DESCRIPTION
**Pull request by amisskii**
Signed-off-by: Anton Misskii <amisskii@redhat.com>

### What does this PR do?

This PR adds a test scenario for the Kubernetes YAML editing feature.

### What issues does this PR fix or reference?

https://github.com/containers/podman-desktop/issues/9259

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  }
+});
+
+test.describe('Kubernetes Edit YAML Feature E2E Test', () => {
+  test.skip(
+    !!process.env.GITHUB_ACTIONS && process.env.RUNNER_OS === 'Linux',
+    'Tests suite should not run on Linux platform',
+  );
+  test('Create a Kubernetes deployment resource', async ({ navigationBar }) => {
+    test.setTimeout(80_000);
```
Comment (axel7083) [inline tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts:83]: is it commun to set the timeout here ?

Might be better to do the following 

```
test('test-name', () => {
    // logic here
}, { 
    timeout: 10_000,
});
```

Comment (cbr7) [reply tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts:83]: We are not using vitest test runner anymore, in PW test runner you set timeout via test.setTimerout call

**Review by jeffmaury — COMMENTED** [review]
Seems this test is coupled with kind what about other clusters ?

**Comment by amisskii** [comment]
> Seems this test is coupled with kind what about other clusters ?

@jeffmaury  I would like to run tests using a Kind cluster, as it is built-in, for a while. After implementing basic e2e tests, for example, for the Minikube extension, we could extend these tests by adding additional possibilities through parameterization or by creating test cases directly in the Minikube extension repository. This approach would also be valid for other clusters. I will create the follow up issue.

File: tests/playwright/resources/kubernetes/test-deployment-resource.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  template:
+    metadata:
+      labels:
+        app: test-deployment-resource
+    spec:
+      containers:
+      - name: test-deployment-resource
+        image: quay.io/podman-desktop-demo/podify-demo-backend:v1
+        ports:
+        - containerPort: 80
```
Comment (odockal) [inline tests/playwright/resources/kubernetes/test-deployment-resource.yaml:21]: please add new line character at the EOF

File: tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+const CLUSTER_CREATION_TIMEOUT: number = 300_000;
+const KIND_NODE: string = `${CLUSTER_NAME}-control-plane`;
+const KUBERNETES_CONTEXT = `kind-${CLUSTER_NAME}`;
+const KUBERNETES_NAMESPACE = 'default';
+const DEPLOYMENT_NAME = 'test-deployment-resource';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const skipKindInstallation = process.env.SKIP_KIND_INSTALL ? process.env.SKIP_KIND_INSTALL : false;
```
Comment (odockal) [inline tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts]: please change the statement into `process.env.SKIP_KIND_INSTALL === 'true'` to is will be boolean.

File: tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    !!process.env.GITHUB_ACTIONS && process.env.RUNNER_OS === 'Linux',
+    'Tests suite should not run on Linux platform',
+  );
+  test('Create a Kubernetes deployment resource', async ({ navigationBar }) => {
+    test.setTimeout(80_000);
+    const podsPage = await navigationBar.openPods();
+    await playExpect(podsPage.heading).toBeVisible();
+    const playYamlPage = await podsPage.openPlayKubeYaml();
+    await playExpect(playYamlPage.heading).toBeVisible();
+    const yamlFilePath = path.resolve(__dirname, '..', '..', 'resources', 'kubernetes', `${DEPLOYMENT_NAME}.yaml`);
```
Comment (odockal) [inline tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts]: move ithe constant out of test method please.

File: tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    await playYamlPage.playYaml(yamlFilePath, {
```
Comment (odockal) [inline tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts]: if the object passed into playYaml can be constant, please do it. It is not repeated now, but constant could be defined in the file.

File: tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    await navigationBar.openKubernetes();
+    await kubernetesBar.openTabPage(KubernetesResources.Deployments);
+    await playExpect(deploymentsPage.heading).toBeVisible();
+    await playExpect(deploymentsPage.getResourceRowByName(DEPLOYMENT_NAME)).toBeVisible();
+    await deploymentsPage.openResourceDetails(DEPLOYMENT_NAME, KubernetesResources.Deployments);
+    await playExpect(deploymentDetails.heading).toBeVisible();
+    await playExpect
+      .poll(async () => deploymentDetails.getState(), { timeout: 80_000 })
+      .toEqual(KubernetesResourceState.Running);
+  });
```
Comment (odockal) [inline tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts:130]: Could you also add cleaning up of the resources you have created?

File: tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    const kubernetesBar = await navigationBar.openKubernetes();
+    const deploymentsPage = await kubernetesBar.openTabPage(KubernetesResources.Deployments);
+    await playExpect(deploymentsPage.heading).toBeVisible();
+    await playExpect(deploymentsPage.getResourceRowByName(DEPLOYMENT_NAME)).toBeVisible();
+    const deploymentDetails = await deploymentsPage.openResourceDetails(
+      DEPLOYMENT_NAME,
+      KubernetesResources.Deployments,
+    );
+    await playExpect(deploymentDetails.heading).toBeVisible();
+    await deploymentDetails.editKubernetsYamlFile('replicas: 3', 'replicas: 5');
```
Comment (odockal) [inline tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts:117]: I am curious what will happen when you try to replace all if it (and probably will) find more that one occurrence of the string to replace?
I think there is more than one `*replicas: 3` string in the file? Can you please confirm?

**Review by odockal — CHANGES_REQUESTED** [review]

File: tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts (hunk 7 above)
Comment (amisskii) [reply tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts:117]: Yes, there are more `*replicas: 3`. However, other than the `spec.replicas` are just metadata, so editing them doesn't affect anything.

Comment (amisskii) [reply tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts:117]: They will still be changed to `*replicas: 5` after applying changes only to `spec.replicas`

**Review by odockal — APPROVED** [review]

**Comment by odockal** [comment]
LGTM. @amisskii Can you please also run on windows wsl?

**Comment by amisskii** [comment]
> LGTM. @amisskii Can you please also run on windows wsl?

Of course, after https://github.com/containers/podman-desktop/issues/9522 is resolved :)

**Comment by odockal** [comment]
@amisskii How is it related?

**Comment by amisskii** [comment]
https://github.com/containers/podman-desktop-e2e/actions/runs/11480847036

**Pull request merged**